File: lib/detectors/firebase.js
module.exports = {
  name: 'Firebase Config',
  severity: 'high',
  run(content) {
    const out = [];
    // Look for typical firebase config object or apiKey
    const apiKeyRe = /apiKey\s*[:=]\s*["'](AIza[0-9A-Za-z\-_]{35})["']/i;
    const cfgRe = /firebase(?:Config)?\s*[:=]\s*\{([\s\S]{0,800}?)};/i;

    const m1 = apiKeyRe.exec(content);
    if (m1) {
      out.push({ value: m1[1], context: content.substr(Math.max(0, m1.index - 40), 120).replace(/\n/g, ' '), severity: 'high' });
    }

    const m2 = cfgRe.exec(content);
    if (m2) {
      const snippet = m2[0].replace(/\n/g, ' ');
      out.push({ value: 'firebaseConfig', context: snippet, severity: 'high' });
    }

    return out;
  }
};

File: JSX/detectors/firebase.py
"""Firebase configuration detector."""

import re
from .base import BaseDetector


class Detector(BaseDetector):
    """Detects Firebase configuration in JavaScript."""

    def __init__(self):
        super().__init__("Firebase Config", severity="high")
        self.api_key_pattern = re.compile(r'apiKey\s*[:=]\s*["\']?(AIza[0-9A-Za-z\-_]{35})["\']?', re.IGNORECASE)
        self.config_pattern = re.compile(r'firebase(?:Config)?\s*[:=]\s*\{([\s\S]{0,800}?)};', re.IGNORECASE)

    def run(self, content):
        """Find Firebase configuration."""
        findings = []

        for match in self.api_key_pattern.finditer(content):
            value = match.group(1)
            context = self._get_context(content, match.start())
            findings.append({
                "value": value,
                "context": context,
                "pos": match.start(),
                "severity": "high"
            })

        for match in self.config_pattern.finditer(content):
            snippet = match.group(0).replace("\n", " ")
            findings.append({
                "value": "firebaseConfig",
                "context": snippet[:200],
                "pos": match.start(),
                "severity": "high"
            })

        return findings
